### ISSUE TITLE
why?????loss = loss_CE + loss_distill.sum() / batch_size / 10000

### ISSUE THREAD

**Issue by jianbaba** [clovaai/overhaul-distillation#12]
hello，would you tell me why is 10000 diveded？thanks

**Comment by bhheo** [comment]
Hi

It is a hyper parameter of distillation which is defined as ``alpha`` in my paper.
Because scale of cross-entropy loss and distillation loss is quite different, 
most of feature distillation needs this hyper parameter to match the scale of losses.

We found that parameter empirically by parameter search on 10 log-scale grid.

**Comment by jianbaba** [comment]
> Hi
> 
> It is a hyper parameter of distillation which is defined as `alpha` in my paper.
> Because scale of cross-entropy loss and distillation loss is quite different,
> most of feature distillation needs this hyper parameter to match the scale of losses.
> 
> We found that parameter empirically by parameter search on 10 log-scale grid.

thanks very much!
Didn't you do some experiments with shufflenetv2! I tried it, however it didn't work.

**Comment by bhheo** [comment]
I'm sorry but, I didn't try on shufflenetV2.
But, I can give you some advice based on my experience.
 
When architecture or size of teacher-student is very different, 
distillation is going to be hard work to student network.
So, I recommend you to use smaller ``alpha`` parameter, such as ``/100000``.
In case of MobileNetV2, I could achieve improvement  based on smaller ``alpha`` parameter.

It might give you some improvement, if distillation didn't work.
But, I also admit that distillation on recent light-weight model is not good as ResNet.

**Comment by jianbaba** [comment]
> I'm sorry but, I didn't try on shufflenetV2.
> But, I can give you some advice based on my experience.
> 
> When architecture or size of teacher-student is very different,
> distillation is going to be hard work to student network.
> So, I recommend you to use smaller `alpha` parameter, such as `/100000`.
> In case of MobileNetV2, I could achieve improvement based on smaller `alpha` parameter.
> 
> It might give you some improvement, if distillation didn't work.
> But, I also admit that distillation on recent light-weight model is not good as ResNet.

감사합니다

**Comment by 94FLY** [comment]
@bhheo Hello！I use this method in OCR. The distill loss is twice as much as cross-entropy loss at the beginning. Should I modify the alpha parameter to make them almost equal?

**Comment by bhheo** [comment]
@94FLY hi
I couldn't find any correlation btw the loss value and proper alpha parameter. So, I don't know it is good or not.
In my experience, large alpha ruins the performance and small alpha reduces the effect of distillation. So, If the performance is not bad, I think you don't need to reduce alpha. But, you have to tune alpha value for the best performance.

**Comment by 94FLY** [comment]
Tanks!

**Comment by 94FLY** [comment]
Thanks!
